### PR TITLE
Bump TF version to 1.12

### PR DESCRIPTION
**Pull request by RaphaelMeudec**
### Summary

Retry for https://github.com/keras-team/keras/pull/11672

Having some trouble with `ctc_decode_beam_search`: I've explored the computed values by Tensorflow `gen_ctc_ops.ctc_beam_search_decoder` and the returned `log_probabilities` are different between 1.10 and 1.11. Investigating on this atm.

### PR Overview

- [ ] This PR requires new unit tests [y/n] (make sure tests are included)
- [ ] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [ ] This PR is backwards compatible [y/n]
- [ ] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by RaphaelMeudec** [comment]
This difference in values comes from a normalization step which has been added to the `ctc_beam_search_decoder` (see [this TF commit](https://github.com/tensorflow/tensorflow/commit/5e07ab3ff51b06c3291789ac0e65499217f720d1)). I'm considering taking the [beam search test values](https://github.com/tensorflow/tensorflow/blob/92985e67564c119203f0aaca94b5a334d87c35e4/tensorflow/python/kernel_tests/ctc_decoder_ops_test.py#L176) from TF, looks good to you?

**Comment by gabrieldemarmiesse** [comment]
I think it's safe to be in sync with tf.keras.

**Comment by gabrieldemarmiesse** [comment]
Thanks for working on it, I'll close my pr.

**Comment by RaphaelMeudec** [comment]
@gabrieldemarmiesse I've an issue with the implementation of ctc_decode. We apply the `tf.log` on the inputs [here](https://github.com/keras-team/keras/blob/1d81a20292ca6926e595d06a6cd725dbb104a146/keras/backend/tensorflow_backend.py#L4504), however it's not the case in Tensorflow. Is it intentional to have this slight difference of behaviour? I've commented the log for now, but if it's deliberate, the other option is to take the exp of the inputs in the tests. What's your opinion on this?

**Comment by gabrieldemarmiesse** [comment]
> We apply the tf.log on the inputs here, however it's not the case in Tensorflow.

Looking at this line: https://github.com/tensorflow/tensorflow/blob/2f7e56a65104aeca0a7fcc98a7b10fb2cff9b1a3/tensorflow/python/keras/backend.py#L5139 it seems that the log is there in tensorflow too. Am I missing something obvious?

**Comment by RaphaelMeudec** [comment]
@gabrieldemarmiesse Indeed, I must have missed this obvious line. Will dig into it and update the PR

**Comment by RaphaelMeudec** [comment]
@gabrieldemarmiesse In the last commit, I got back to the correct behaviour (log of the inputs). In the test, I took the exponential of the inputs because:

- in TF, the test only applies to `ctc_decode_beam_search` and not `keras.backend.ctc_decode`, so the log is not applied in this case
- in Keras, the test applies on `ctc_decode` so the log is applied

Taking the exp of the inputs is the only way to take the same inputs values for testing.

Similar behaviour is happening in the greedy version:

- in TF, the `testCTCGreedyDecoder` applies the log to the inputs in the test and `ctc_decode_greedy` is applied directly to the inputs
- in Keras, the `test_ctc_decode_greedy` doesn't apply the log, but it is applied in the `ctc_decode` method

Hope it's kind of clear, let me know if you need some precisions!

**Comment by RaphaelMeudec** [comment]
@gabrieldemarmiesse Don't know why the Py3.6 TF timed out on CI, can you force a re-run on it?

**Comment by RaphaelMeudec** [comment]
@gabrieldemarmiesse It seems all Tensorflow flows are passing, only CNTK times out. Don't know what the problem is as this PR only tackles Tensorflow elements

**Comment by gabrieldemarmiesse** [comment]
I wonder if the tensorflow upgrade is actually what makes stuff time out. Since tensorflow is used in the cntk build (for tensorboard with cntk for example), there may be a memory leak somewhere. We definitly need to go to the bottom of it. 

Let's wait before merging this. If this is TF causing the timeouts, we don't want it in master, we'll end up with PRs failing all the time. I'll do some investigation.

Thanks for your work anyway. It was very useful. I hope we merge it soon.

**Comment by RaphaelMeudec** [comment]
@gabrieldemarmiesse It seems that cntk backend already times out on master since the tf.keras API sync merged by @fchollet  See: https://travis-ci.org/keras-team/keras/builds/477602449
If you do investigate on the matter, you might want to start with this merge commit https://github.com/keras-team/keras/compare/6096ded90df9...1336cdb14ff0

**Comment by gabrieldemarmiesse** [comment]
Thanks for the heads up!
